lib/galaxy/tool_util/parser/xml.py

```diff
@@ -499,9 +499,14 @@ def parse_help(self):
         help_elem = self.root.find('help')
         return help_elem.text if help_elem is not None else None
 
+    @property
     def macro_paths(self):
         return self._macro_paths
 
+    @property
+    def source_path(self):
+        return self._source_path
```

Comment:
Can this be the default implementation in `ToolSource()` ? Or copy this to `CwlToolSource` and `YamlToolSource`.